puzzles/models.py

```diff
@@ -35,9 +37,17 @@ def set_answer(self, answer):
         self.status = Puzzle.SOLVED
         self.save()
 
-    def clear_answer(self):
+    def clear_answer(self, guess):
+        # puzzle already solved by different guess, so this current guess does not affect state
+        if self.status == Puzzle.SOLVED and self.answer != guess:
+            return
+
         self.answer = ''
-        self.status = Puzzle.SOLVING
+        if self.guesses.filter(Q(status=Answer.NEW) | Q(status=Answer.SUBMITTED)):
```

Comment:
Where is guesses defined?

Reply:
I was wondering about this, too, and apparently it's a reverse mapping defined by the `related_name` field in `answers/model.py`:

```
class Answer(models.Model):
    ...
    puzzle = models.ForeignKey('puzzles.Puzzle', on_delete=models.CASCADE, related_name="guesses")
```

I think this means that given a `Puzzle`, you can retrieve all `Answer`s that have been submitted for the puzzle by using the `guesses` field.

Ref: https://stackoverflow.com/questions/2642613/what-is-related-name-used-for-in-django

Reply:
Ah thanks for explaining.